### PR TITLE
Defect for SQL schema information exposure on error

### PR DESCRIPTION
**Pull request by RoopGuron**
- Defect for SQL schema information exposure on error

### CODE REVIEW

**Comment by 6palace** [comment]
I notice that jdbcTemplate has an ExceptionTranslator property, and the same jdbcTemplate is used by all our jdbc services, maybe instantiating an Exception translator might be a more thorough fix for this defect?

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ScimUserEndpointsIntegrationTests.java [hunk 1]
```diff
@@ -140,6 +143,22 @@ public void createUserSucceeds() throws Exception {
         assertTrue(joe2.isVerified());
     }
 
+    // curl -v -H "Content-Type: application/json" -H "Accept: application/json"
```
Comment (arjunnayak) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ScimUserEndpointsIntegrationTests.java]: remove comments?

Comment (RoopGuron) [reply uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ScimUserEndpointsIntegrationTests.java]: I believe those are helpful in troubleshooting .....it is the test code and given the beast that uaa...any help and directions is good...

**Comment by RoopGuron** [comment]
A more thorough fix would be to Replace the `SQLErrorCodeSQLExceptionTranslator` with a custom implementation and configure that with jdbcTemplate. The custom implementation must : 

- **Replace** the error message completely(optionally logging the original message to the logs).
- Does **not** set the cause/rootcause exception to the mapped/translated exception.

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ScimUserEndpointsIntegrationTests.java (hunk 1 above)
Comment (arjunnayak) [reply uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/ScimUserEndpointsIntegrationTests.java]: good point, thanks

**Comment by arjunnayak** [comment]
created story in backlog to improve/extend the spring `SQLErrorCodeSQLExceptionTranslator`

File: server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java [hunk 2]
```diff
@@ -210,7 +211,12 @@ public void setValues(PreparedStatement ps) throws SQLException {
             userDetails.put("verified", existingUser.isVerified());
             userDetails.put("user_id", existingUser.getId());
             throw new ScimResourceAlreadyExistsException("Username already in use: " + existingUser.getUserName(), userDetails);
+        } catch(DataIntegrityViolationException e) {
+            logger.debug("DataIntegrityViolationException thrown: ", e);
+                throw new InvalidScimResourceException("ScimUser:" + user + " is Invalid and can not be created. " +
+                        "Please validate the property values.");
```
Comment (Bitty9) [inline server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java]: user should be user.toString() for the exception message. Also fix the capitalization for the message.

Comment (RoopGuron) [reply server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java]: user.toString() is always implicitly called in string concatenation... 
capitalization is fine I believe. Invalid is capital for emphasis.

Comment (Bitty9) [reply server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java]: "ScimUser:org.cloudfoundry.identity.uaa.scim.ScimUser@7a04fea7 is Invalid and can not be created. Please validate the property values." is what is being printed out so needs to be fixed.

Comment (RoopGuron) [reply server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java]: In that case ScimUser needs to have a toString implementation.  Should be added at the earliest.

Comment (arjunnayak) [reply server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java]: Updated to be:
"ScimUser:<username_submitted> is invalid and can not be created. Please validate the property values for size and type."

**Pull request merged**